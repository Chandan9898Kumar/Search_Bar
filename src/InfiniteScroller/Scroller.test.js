import { render, screen,fireEvent } from "@testing-library/react";
import MainComponent from "./Scroller";
import Enzyme from "enzyme";
import { shallow, mount } from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";
Enzyme.configure({ adapter: new Adapter() });

describe("Check Main Scroller Component", () => {
  it("Component Should be defined", () => {
    const wrapper = mount(<MainComponent />);
    expect(wrapper).toBeDefined();
  });

  it('Should check Input Field',()=>{
    render(<MainComponent />);
    const fireInput=screen.getByTestId('searchField')
    fireEvent.change(fireInput,{target:{value:'search'}})
  })
  it('should check div element',()=>{
    const { container }= render(<MainComponent />);
    expect(container).toMatchSnapshot();
  })
});
