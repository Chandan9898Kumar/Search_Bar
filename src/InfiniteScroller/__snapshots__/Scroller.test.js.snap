// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Main Scroller Component should check div element 1`] = `
<div>
  <div
    class="containerOfScroller"
  >
    <div
      class="ScrollText"
    >
      <span>
        Infinite Scroller
      </span>
    </div>
    <div
      class="ScrollInput"
    >
      <svg
        class="iconOfSearch"
        fill="currentColor"
        height="1em"
        id="searchIcon"
        stroke="currentColor"
        stroke-width="0"
        style="color: gray;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
      <input
        class="scrollInputText"
        data-testid="searchField"
        placeholder="Search Here..."
        type="search"
        value=""
      />
    </div>
    <div>
      <div>
        Loading
      </div>
    </div>
  </div>
</div>
`;
